merkle-tree/src/mmcs.rs

```diff
@@ -104,6 +117,28 @@ where
         opened_values: &[Vec<P::Scalar>],
         proof: &Self::Proof,
     ) -> Result<(), Self::Error> {
+        // Check that the openings have the correct shape.
+        if dimensions.len() != opened_values.len() {
+            return Err(WrongBatchSize);
+        }
+        // TODO: Disabled for now since TwoAdicFriPcs and CirclePcs currently pass 0 for width.
+        // for (dims, opened_vals) in dimensions.iter().zip(opened_values) {
+        //     if opened_vals.len() != dims.width {
+        //         return Err(WrongWidth);
+        //     }
+        // }
+        let max_height = dimensions.iter().map(|dim| dim.height).max().unwrap();
+        let log_max_height = log2_ceil_usize(max_height);
+        dbg!(dimensions);
+        dbg!(proof.len());
```

Comment:
@rjeli when we use the circle PCS, it seems these two values sometimes match but sometimes don't, e.g. on `prove_m31_circle_deg2` I see

```
[merkle-tree/src/mmcs.rs:132:9] dimensions = [
    0x512,
]
[merkle-tree/src/mmcs.rs:133:9] proof.len() = 9
[merkle-tree/src/mmcs.rs:132:9] dimensions = [
    0x512,
    0x512,
]
[merkle-tree/src/mmcs.rs:133:9] proof.len() = 9
[merkle-tree/src/mmcs.rs:132:9] dimensions = [
    0x512,
]
[merkle-tree/src/mmcs.rs:133:9] proof.len() = 8
```

It's weird, I thought the heights would all be uniform until FRI folding (and I believe this is still input layer, before folding). Can you can think of a reason the proof length would change like that?